android/intltest/java/com/facebook/hermes/test/HermesIntlAndroidTest.java

```diff
@@ -80,4 +85,46 @@ public void testDateTimeFormatCaseInsensitivity() {
       assertThat(result).isEqualTo("9/24, 6:00 PM");
     }
   }
+
+  @Test
+  public void testSignDisplayAlwaysForApiLevelAbove31() {
+    if (Build.VERSION.SDK_INT >= 31) {
+      try (JSRuntime rt = JSRuntime.makeHermesRuntime()) {
+        rt.evaluateJavaScript(
+            "var nf = new Intl.NumberFormat('en-US', { signDisplay: 'always' });\n" +
+            "var result = nf.format(123);");
+
+        String result = rt.getGlobalStringProperty("result");
+        assertThat(result).isEqualTo("+123"); // Adjust expected output according to your logic
+      }
+    }
+  }
+
+  @Test
+  public void testSignDisplayNeverForApiLevelAbove31() {
+    if (Build.VERSION.SDK_INT >= 31) {
+      try (JSRuntime rt = JSRuntime.makeHermesRuntime()) {
+        rt.evaluateJavaScript(
+            "var nf = new Intl.NumberFormat('en-US', { signDisplay: 'never' });\n" +
+            "var result = nf.format(123);");
+
+        String result = rt.getGlobalStringProperty("result");
+        assertThat(result).isEqualTo("123");
+      }
+    }
+  }
+
+  @Test
+  public void testSignDisplayExceptZeroForApiLevelAbove31() {
+    if (Build.VERSION.SDK_INT >= 31) {
+      try (JSRuntime rt = JSRuntime.makeHermesRuntime()) {
+        rt.evaluateJavaScript(
+            "var nf = new Intl.NumberFormat('de-DE', { exceptZero: 'never', currency: 'EUR' });\n" +
```

Comment:
This test seems odd, is it correct? I'm not aware of `exceptZero` being an option to `NumberFormat` (as opposed to `signDisplay: "exceptZero"`), and I don't believe the currency is printed unless you set `style: "currency"`.

If this is incorrect, we should understand why this test didn't fail.

Reply:
I've corrected the test case to properly reflect the intended behaviour by using signDisplay: 'exceptZero' and including the style: 'currency' option.